release/create-release-candidate-branch.sh

```diff
@@ -344,10 +344,7 @@ main() {
 	check_dependencies
 
 	# sanity checks before we start: folder, branches etc.
-	# deactivate -e so that piped commands can be used
-	set +e
```

Comment:
This short-circuits without logging out helpful information. e.g. running this in main

> ./release/create-release-candidate-branch.sh -t 25.7.0-rc1 -w products

shows:

> Expected release branch is missing: release-25.7

which is suppressed with this PR change, so I am not sure this is an improvement.

Reply:
Although we certainly exit more consistently now, albeit with sometimes less info.